internal/analyzers/prometheus.go

```diff
@@ -0,0 +1,271 @@
+	}
+
+	slog.Info("Prometheus is compliant, no issues found", "name", name, "namespace", namespace)
+	return nil
+}
+
+func checkClusterRoleRules(crb v1.ClusterRoleBinding, cr *v1.ClusterRole) error {
+	var errs []string
+	verbsToCheck := []string{"get", "list", "watch"}
+	missingVerbs := []string{}
+
+	for _, rule := range cr.Rules {
+		for _, resource := range rule.Resources {
+			found := false
+			if resource == "configmaps" {
+				for _, verb := range rule.Verbs {
+					if verb == "get" {
+						found = true
+						break
+					}
+				}
+				if !found {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'configmaps' with 'get' in its verbs", crb.RoleRef.Name))
+				}
+				continue
+			}
+			for range rule.APIGroups {
+				for _, requiredVerb := range verbsToCheck {
+					found := false
+					for _, verb := range rule.Verbs {
+						if verb == requiredVerb {
+							found = true
+							break
+						}
+					}
+					if !found {
+						missingVerbs = append(missingVerbs, requiredVerb)
+					}
+				}
+				if len(missingVerbs) > 0 {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s is missing necessary verbs for APIGroups: %v", crb.RoleRef.Name, missingVerbs))
+				}
+			}
+		}
+		for _, nonResource := range rule.NonResourceURLs {
+			if nonResource == "/metrics" {
+				hasGet := false
+				for _, verb := range rule.Verbs {
+					if verb == "get" {
+						hasGet = true
+						break
+					}
+				}
+				if !hasGet {
+					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'get' verb for NonResourceURL '/metrics'", crb.RoleRef.Name))
+				}
+			}
+		}
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("multiple errors found:\n%s", strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func checkPrometheusNamespaceSelectorsStatus(ctx context.Context, clientSets *k8sutil.ClientSets, labelSelector *metav1.LabelSelector) (bool, error) {
+    if labelSelector == nil {
+        return true, nil
+    }
+
+    if len(labelSelector.MatchLabels) == 0 && len(labelSelector.MatchExpressions) == 0 {
+        _, err := clientSets.KClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+        if err != nil {
+            return false, fmt.Errorf("Failed to list Namespaces: %v", err)
+        }
+        return false, nil
+    }
+
+    labelMap, err := metav1.LabelSelectorAsMap(labelSelector)
+    if err != nil {
+        return false, fmt.Errorf("Invalid label selector format: %v", err)
+    }
+
+    nsSelector := labels.SelectorFromSet(labelMap)
+
+    options := metav1.ListOptions{
+        LabelSelector: nsSelector.String(),
+    }
+
+    namespaces, err := clientSets.KClient.CoreV1().Namespaces().List(ctx, options)
+    if err != nil {
+        return false, fmt.Errorf("Failed to list namespaces: %v", err)
+    }
+
+    if len(namespaces.Items) == 0 {
+        return false, fmt.Errorf("No Namespaces match the provided selector")
+    }
+
+    matchedNamespaces := []string{}
```

Comment:
We don't need to print what are the fetched namespaces, we just care if the selectors are matching any namespace.

Reply:
done